### PR TITLE
adding category wise page for items

### PR DESCRIPTION
**Pull request by rajaranjanmr**
adding the category wise section , routes for the ecom app.Adding on the same branch

### CODE REVIEW

File: src/Main.js [hunk 1]
```diff
@@ -7,18 +7,20 @@ import { Link } from "react-router-dom";
...
       <Routes>
         <Route path="/" element={<Home />} />
         <Route path="/cart" element={<Cart />} />
         <Route path="/wishlist" element={<Wishlist />} />
         <Route path="/logout" element={<Logout />} />
         <Route path="/productlisting" element={<ProductListing/>} />
+        <Route path="/mockman" element={<MockAPI/>} />
       </Routes>
-    </div> 
+      </>
```
Comment (nikhiljugale007) [inline src/Main.js:24]: suggestion: You can provide these routes directly in app.js itself. No need to create a Main.js component.

File: src/components/homepage-category-cart.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+      </div>
+    <h1>{props.value.categoryName}</h1>
+    <p className="price">{props.value.priceRange}</p>
+    <p>{props.value.description}</p>
+    <p><Link to="/ProductListing" onClick={()=>productPageDispatch({type: 'CATEGORY_NAME', payload: props.value.categoryName})}>more</Link></p>
+  </div>
+  
+    )
+}
```
Comment (nikhiljugale007) [inline src/components/homepage-category-cart.js:25]: suggestion: You should format (using prettier) your code before pushing. It makes the code more readable.

File: src/components/navigation.css [hunk 3]
```diff
@@ -93,11 +95,11 @@
       width:100%;
       flex-wrap: wrap-reverse;
         /* z-index:0; */
-      
+/*       
      position: absolute;
     top:10rem;
 
-  }
+  } */ */
```
Comment (nikhiljugale007) [inline src/components/navigation.css:102]: try removing commented/ unused code.

File: src/components/rating.css [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    left:30%;
+    text-align:center;
+  }
+  
+  input[type="radio"] {
+    display:none;
+  }
+  
+  .star-rating label {
+    color:#ccc;
```
Comment (nikhiljugale007) [inline src/components/rating.css:20]: fix:you should use color variables

File: src/pages/Home.js [hunk 5]
```diff
@@ -27,9 +28,45 @@ export default function Home(){
...
+  //       } catch (error) {
+  //         setStatus("error");
+  //         setError(error);
+  //       }
+  //     }
+  //     callAsync(msg);
+  //   }, [msg]);
+  
+  //   return { data, status, error };
+  // };
```
Comment (nikhiljugale007) [inline src/pages/Home.js:47]: remove commented code.

File: src/components/clhome.css [hunk 6]
```diff
@@ -253,4 +253,16 @@ iframe{
...
+.button-container{
+    padding: 5rem;
+    position: relative;
+    left: 39%;
+    top: 60%;
+    display: flex;
+    gap: 1em;
+    width: 24%;
+    text-align: center;
+    background-color: aliceblue;
```
Comment (AshutoshM29) [inline src/components/clhome.css:265]: You should use var(color) here as it will make your code consistent

File: src/backend/db/products.js [hunk 7]
```diff
@@ -0,0 +1,256 @@
...
+      "best biryanis , that you never tasted",
+      rating:"4",
+      type:"vanila"
+  },
+  {
+    _id: uuid(),
+    price: "₹350 1pound",
+    item:"cakes",
+    categoryName: "cakes",
+    category:"cakes",
```
Comment (AshutoshM29) [inline src/backend/db/products.js:248]: I'm not sure why are you using these 2??

**Review by AshutoshM29 — APPROVED** [review]
Work on the suggested changes by me and nikhil, rest looks good to merge!

**Pull request merged**